wordpress-mobile/WordPress-Android
# Track site_created event

~`Not-ready-for-review` -> I'd like to wait for an input from @aerych as I haven't been following the related discussions. Eric, if you agree with the proposed solution, please remove the "Not Ready for Review" label. If you have some comments, let me know and I'll make any necessary changes;). Thanks!~

Fixes #9959 

I've created two new events
1.  ENHANCED_SITE_CREATION_PREVIEW_OK_BUTTON_TAPPED - emitted when the user presses "OK" button on the Site Preview screen (previously called "ENHANCED_SITE_CREATION_COMPLETED")

2. SITE_CREATED - emitted when we believe the site has been successfully created
    a. We've received a successful response from the server
    b. We've received `SITE_NAME_EXISTS` error, but we are in retry mode -> we assume the site was already created, but the response from the server got lost. This assumption might be wrong in extreme edge cases, but we decided to live with it as implementing bulletproof solution would be too much effort for too little gain.


Note 2b. The "site_created" event wasn't being recorded in this case in the old site creation flow. However, I believe we want to record it to make sure the number of recorded events is as close to the number of created sites as possible.

To test:
My -> App Settings -> Privacy Settings -> Check Collect information switch

1. Start the Site Creation flow
2. Open LogCat
3. Finish the flow and notice LogCat contains `Tracked: site_created` msg
4. Click on the ok button on the SitePreview screen and notice LogCat contains `Tracked: enhanced_site_creation_preview_ok_button_tapped`
----------
1. Start the Site Creation flow
2. Open LogCat
3. Turn on airplane mode right after you pick a domain and click on the "Create Site" button
4. Notice an error screen is shown
5. Turn off airplane mode
6. Click on Retry  and notice LogCat contains `Tracked: site_created` msg
-----------
1. Start the Site Creation flow
2. Open LogCat
3. Press home button (put the app into background) right after you pick a domain and click on the "Create Site" button
4. Notice LogCat contains `Tracked: site_created` msg

@aerych Can you please confirm this all makes sense. Thanks

cc @diegoreymendez As we'll probably want to preserve cross-platform parity.

Update release notes:

- There are no user facing changes.
